pkg/pidfile/pidfile.go

```diff
@@ -51,20 +54,25 @@ func Write(path string) error {
 		return err
 	}
 
-	// Handle the cleanup
-	sig := make(chan os.Signal, 1)
-	signal.Notify(sig, syscall.SIGQUIT, syscall.SIGINT, syscall.SIGHUP, syscall.SIGTERM)
-	go func() {
-		for s := range sig {
-			log.WithField("signal", s).Info("Exiting due to signal")
-			Remove(path)
-			os.Exit(0)
-		}
-	}()
+	HandleCleanup(cleanUPWg, cleanUPSig, func() {
+		Remove(path)
+	})
 
 	return nil
 }
 
+// Clean cleans up everything created by this package. It closes the returned
+// channel once everything is cleaned up.
+func Clean() <-chan struct{} {
```

Comment:
I think we need to also call this from cilium-health. It currently uses pidfile.Write but now doesn't automatically get the cleanup goroutines.